lib/src/data/course.dart

```diff
@@ -0,0 +1,29 @@
+import "utils.dart";
+import "chat.dart";
+
+extension type CourseId(String value) {}
+
+class Course {
+  final String department;
+  final int course_number;
+  final String subject;
+  Chat chat;
+ 
+  Course({
+	required this.CRN,
```

Comment:
Note that we're not using CRN's for courses anymore, just regular course names like `WRIT 111`. You can make a helper for this: 

```dart
CourseID get id => CourseID("$department $courseNumber");
```

And use that logic in your code